sdk/monitor/azure-monitor-opentelemetry/tests/test_constants.py

```diff
@@ -135,6 +135,7 @@ def test_env_var_or_default_empty_with_defaults(self):
         return_value=True,
     )
     def test_attach_enabled(self, mock_isdir):
+        _constants._IS_ON_APP_SERVICE = True
```

Comment:
Need to reset state after test is done since you are changing a module level variable.